packages/custom-elements-json-core/src/utils/index.ts

```diff
@@ -172,12 +185,20 @@ export function hasAggregatingImport(node: any): boolean {
 }
 
 export function isBareModuleSpecifier(specifier: string): boolean {
-  return !!(specifier.replace(/'/g, ''))[0].match(/[a-zA-Z]/g);
+  return !!specifier.replace(/'/g, '')[0].match(/[a-zA-Z]/g);
 }
 
 export interface Import {
   name: string;
-  kind: "default" | "named" | "aggregate",
+  kind: 'default' | 'named' | 'aggregate' | 'javascript-module';
   importPath: string;
-  isBareModuleSpecifier: boolean
-}
\ No newline at end of file
+  isBareModuleSpecifier: boolean;
+}
+
+/** Type asserters */
+export function isImport<T>(C: T | Import): C is Import {
```

Comment:
My typescript-foo obviously isnt very strong, could you provide some details on this syntax and how it works exactly?

`isImport<T>(C: T | Import): C is Import {`

Reply:
Basically what it does is to take in an argument which can be of type Import or some thing else (T). And then we declare that based on the boolean we return in the function we can assert wether or not C is of type Import. 

So whenever we use this in a if-statement typescript will understand that the variable in question is in fact of type Import.

Reply:
TIL, thanks!